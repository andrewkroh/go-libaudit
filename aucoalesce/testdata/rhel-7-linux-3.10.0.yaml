---
tests:
  changed-system-time: |
    type=SYSCALL msg=audit(1492798541.391:20246): arch=c000003e syscall=159 success=yes exit=0 a0=7f71508da920 a1=1 a2=0 a3=7f715258624c items=0 ppid=1 pid=1075 auid=4294967295 uid=38 gid=38 euid=38 suid=38 fsuid=38 egid=38 sgid=38 fsgid=38 tty=(none) ses=4294967295 comm="ntpd" exe="/usr/sbin/ntpd" subj=system_u:system_r:ntpd_t:s0 key="time-change"
    type=EOE msg=audit(1492798541.391:20246):

  make-device: |
    type=SYSCALL msg=audit(1492800799.050:20294): arch=c000003e syscall=133 success=yes exit=0 a0=7ff61dde1960 a1=1180 a2=0 a3=2 items=2 ppid=1 pid=326 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="systemd-logind" exe=2F7573722F6C69622F73797374656D642F73797374656D642D6C6F67696E643B3538663938633835202864656C6574656429 subj=system_u:system_r:systemd_logind_t:s0 key="specialfiles"
    type=CWD msg=audit(1492800799.050:20294):  cwd="/"
    type=PATH msg=audit(1492800799.050:20294): item=0 name="/run/systemd/sessions/" inode=11378 dev=00:13 mode=040755 ouid=0 ogid=0 rdev=00:00 obj=system_u:object_r:systemd_logind_sessions_t:s0 objtype=PARENT
    type=PATH msg=audit(1492800799.050:20294): item=1 name="/run/systemd/sessions/23.ref" inode=98040 dev=00:13 mode=010600 ouid=0 ogid=0 rdev=00:00 obj=system_u:object_r:systemd_logind_sessions_t:s0 objtype=CREATE
    type=EOE msg=audit(1492800799.050:20294):

  CONFIG_CHANGE backlog limit: |
    type=CONFIG_CHANGE msg=audit(1492753795.844:15406): audit_backlog_limit=64 old=8192 auid=4294967295 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 res=1

  CONFIG_CHANGE add rule: |
    type=CONFIG_CHANGE msg=audit(1492749467.018:1209): auid=4294967295 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 op="add_rule" key="pam" list=4 res=1

  LOGIN success: |
    type=LOGIN msg=audit(1481077043.057:414): pid=1298 uid=0 subj=system_u:system_r:sshd_t:s0-s0:c0.c1023 old-auid=4294967295 auid=1000 old-ses=4294967295 ses=1 res=1

  USER_LOGIN: |
    type=USER_LOGIN msg=audit(1493046015.752:42120): pid=18981 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:sshd_t:s0-s0:c0.c1023 msg='op=login acct="root" exe="/usr/sbin/sshd" hostname=? addr=46.160.144.250 terminal=ssh res=failed'
