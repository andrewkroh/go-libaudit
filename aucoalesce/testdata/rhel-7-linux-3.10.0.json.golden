[
  {
    "test_name": "CONFIG_CHANGE add rule",
    "event": {
      "@timestamp": "2017-04-21T04:37:47.018Z",
      "sequence": 1209,
      "category": "configuration",
      "record_type": "config_change",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "attrs": {
          "auid": "unset"
        },
        "selinux": {
          "domain": "unconfined_service_t",
          "level": "s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "add_rule",
        "what": "audit-config"
      },
      "action": "changed-audit-configuration",
      "key": "pam",
      "data": {
        "list": "4",
        "op": "add_rule"
      }
    }
  },
  {
    "test_name": "CONFIG_CHANGE backlog limit",
    "event": {
      "@timestamp": "2017-04-21T05:49:55.844Z",
      "sequence": 15406,
      "category": "configuration",
      "record_type": "config_change",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "attrs": {
          "auid": "unset"
        },
        "selinux": {
          "domain": "unconfined_service_t",
          "level": "s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "64",
        "what": "audit-config"
      },
      "action": "changed-audit-configuration",
      "data": {
        "audit_backlog_limit": "64",
        "old": "8192"
      }
    }
  },
  {
    "test_name": "CRYPTO_KEY_USER",
    "event": {
      "@timestamp": "2016-12-07T02:17:21.497Z",
      "sequence": 404,
      "category": "crypto",
      "record_type": "crypto_key_user",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "0",
        "attrs": {
          "auid": "unset",
          "suid": "0",
          "uid": "0"
        },
        "selinux": {
          "category": "c0.c1023",
          "domain": "sshd_t",
          "level": "s0-s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "6d:a3:7f:ed:de:4a:79:f2:aa:49:ec:d1:75:36:97:a3",
        "secondary": "96.241.146.97",
        "what": "user-session"
      },
      "action": "negotiated-crypto-key",
      "how": "/usr/sbin/sshd",
      "data": {
        "addr": "96.241.146.97",
        "exe": "/usr/sbin/sshd",
        "fp": "6d:a3:7f:ed:de:4a:79:f2:aa:49:ec:d1:75:36:97:a3",
        "kind": "server",
        "op": "destroy",
        "pid": "1299",
        "spid": "1299"
      }
    }
  },
  {
    "test_name": "CRYPTO_SESSION",
    "event": {
      "@timestamp": "2016-12-07T02:17:21.515Z",
      "sequence": 406,
      "category": "crypto",
      "record_type": "crypto_session",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "0",
        "attrs": {
          "auid": "unset",
          "suid": "74",
          "uid": "0"
        },
        "selinux": {
          "category": "c0.c1023",
          "domain": "sshd_t",
          "level": "s0-s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "96.241.146.97",
        "secondary": "63927",
        "what": "user-session"
      },
      "action": "started-crypto-session",
      "how": "/usr/sbin/sshd",
      "data": {
        "addr": "96.241.146.97",
        "cipher": "chacha20-poly1305@openssh.com",
        "direction": "from-server",
        "exe": "/usr/sbin/sshd",
        "ksize": "512",
        "laddr": "10.142.0.2",
        "lport": "22",
        "op": "start",
        "pfs": "curve25519-sha256@libssh.org",
        "pid": "1298",
        "rport": "63927",
        "spid": "1299"
      }
    }
  },
  {
    "test_name": "LOGIN success",
    "event": {
      "@timestamp": "2016-12-07T02:17:23.057Z",
      "sequence": 414,
      "category": "unknown",
      "record_type": "login",
      "result": "success",
      "session": "1",
      "actor": {
        "primary": "4294967295",
        "secondary": "0",
        "attrs": {
          "auid": "1000",
          "old-auid": "4294967295",
          "uid": "0"
        },
        "selinux": {
          "category": "c0.c1023",
          "domain": "sshd_t",
          "level": "s0-s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "1000",
        "what": "user-session"
      },
      "action": "changed-login-id-to",
      "data": {
        "old-ses": "4294967295",
        "pid": "1298"
      }
    }
  },
  {
    "test_name": "USER_LOGIN",
    "event": {
      "@timestamp": "2017-04-24T15:00:15.752Z",
      "sequence": 42120,
      "category": "user-login",
      "record_type": "user_login",
      "result": "fail",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "root",
        "attrs": {
          "auid": "unset",
          "uid": "0"
        },
        "selinux": {
          "category": "c0.c1023",
          "domain": "sshd_t",
          "level": "s0-s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "ssh",
        "secondary": "46.160.144.250",
        "what": "user-session"
      },
      "action": "logged-in",
      "how": "/usr/sbin/sshd",
      "data": {
        "acct": "root",
        "addr": "46.160.144.250",
        "exe": "/usr/sbin/sshd",
        "op": "login",
        "pid": "18981",
        "terminal": "ssh"
      }
    }
  },
  {
    "test_name": "USER_LOGOUT",
    "event": {
      "@timestamp": "2017-04-26T03:56:45.243Z",
      "sequence": 73336,
      "category": "user-login",
      "record_type": "user_logout",
      "result": "success",
      "session": "132",
      "actor": {
        "primary": "1000",
        "secondary": "1000",
        "attrs": {
          "auid": "1000",
          "uid": "0"
        },
        "selinux": {
          "category": "c0.c1023",
          "domain": "sshd_t",
          "level": "s0-s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "/dev/pts/1",
        "what": "user-session"
      },
      "action": "logged-out",
      "how": "/usr/sbin/sshd",
      "data": {
        "exe": "/usr/sbin/sshd",
        "id": "1000",
        "op": "login",
        "pid": "2370",
        "terminal": "/dev/pts/1"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "changed-system-time",
    "event": {
      "@timestamp": "2017-04-21T18:15:41.391Z",
      "sequence": 20246,
      "category": "audit-rule",
      "record_type": "syscall",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "38",
        "attrs": {
          "auid": "unset",
          "egid": "38",
          "euid": "38",
          "fsgid": "38",
          "fsuid": "38",
          "gid": "38",
          "sgid": "38",
          "suid": "38",
          "uid": "38"
        },
        "selinux": {
          "domain": "ntpd_t",
          "level": "s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "what": "system"
      },
      "action": "changed-system-time",
      "how": "/usr/sbin/ntpd",
      "key": "time-change",
      "data": {
        "a0": "7f71508da920",
        "a1": "1",
        "a2": "0",
        "a3": "7f715258624c",
        "arch": "x86_64",
        "comm": "ntpd",
        "exe": "/usr/sbin/ntpd",
        "exit": "0",
        "items": "0",
        "pid": "1075",
        "ppid": "1",
        "syscall": "adjtimex",
        "tty": "(none)"
      }
    }
  },
  {
    "test_name": "make-device",
    "event": {
      "@timestamp": "2017-04-21T18:53:19.05Z",
      "sequence": 20294,
      "category": "audit-rule",
      "record_type": "syscall",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "0",
        "attrs": {
          "auid": "unset",
          "egid": "0",
          "euid": "0",
          "fsgid": "0",
          "fsuid": "0",
          "gid": "0",
          "sgid": "0",
          "suid": "0",
          "uid": "0"
        },
        "selinux": {
          "domain": "systemd_logind_t",
          "level": "s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "/run/systemd/sessions/",
        "secondary": "11378",
        "what": "file",
        "selinux": {
          "domain": "systemd_logind_sessions_t",
          "level": "s0",
          "role": "object_r",
          "user": "system_u"
        }
      },
      "action": "make-device",
      "how": "/usr/lib/systemd/systemd-logind;58f98c85 (deleted)",
      "key": "specialfiles",
      "data": {
        "a0": "7ff61dde1960",
        "a1": "1180",
        "a2": "0",
        "a3": "2",
        "arch": "x86_64",
        "comm": "systemd-logind",
        "cwd": "/",
        "exe": "/usr/lib/systemd/systemd-logind;58f98c85 (deleted)",
        "exit": "0",
        "pid": "326",
        "ppid": "1",
        "syscall": "mknod",
        "tty": "(none)"
      },
      "paths": [
        {
          "dev": "00:13",
          "inode": "11378",
          "item": "0",
          "mode": "040755",
          "name": "/run/systemd/sessions/",
          "obj_domain": "systemd_logind_sessions_t",
          "obj_level": "s0",
          "obj_role": "object_r",
          "obj_user": "system_u",
          "objtype": "PARENT",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        },
        {
          "dev": "00:13",
          "inode": "98040",
          "item": "1",
          "mode": "010600",
          "name": "/run/systemd/sessions/23.ref",
          "obj_domain": "systemd_logind_sessions_t",
          "obj_level": "s0",
          "obj_role": "object_r",
          "obj_user": "system_u",
          "objtype": "CREATE",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        }
      ]
    }
  },
  {
    "test_name": "mount syscall",
    "event": {
      "@timestamp": "2017-04-25T14:01:01.69Z",
      "sequence": 48922,
      "category": "audit-rule",
      "record_type": "syscall",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "0",
        "attrs": {
          "auid": "unset",
          "egid": "0",
          "euid": "0",
          "fsgid": "0",
          "fsuid": "0",
          "gid": "0",
          "sgid": "0",
          "suid": "0",
          "uid": "0"
        },
        "selinux": {
          "domain": "systemd_logind_t",
          "level": "s0",
          "role": "system_r",
          "user": "system_u"
        }
      },
      "thing": {
        "primary": "/run/user/0",
        "secondary": "454267",
        "what": "file",
        "selinux": {
          "domain": "user_tmp_t",
          "level": "s0",
          "role": "object_r",
          "user": "system_u"
        }
      },
      "action": "mounted",
      "how": "/usr/lib/systemd/systemd-logind;58f98c85 (deleted)",
      "key": "mount",
      "data": {
        "a0": "7ff61c0a7270",
        "a1": "7ff61dddf470",
        "a2": "7ff61c0a7270",
        "a3": "6",
        "arch": "x86_64",
        "comm": "systemd-logind",
        "cwd": "/",
        "exe": "/usr/lib/systemd/systemd-logind;58f98c85 (deleted)",
        "exit": "0",
        "pid": "326",
        "ppid": "1",
        "syscall": "mount",
        "tty": "(none)"
      },
      "paths": [
        {
          "item": "0",
          "name": "/run/user/0",
          "objtype": "UNKNOWN"
        },
        {
          "dev": "00:13",
          "inode": "454267",
          "item": "1",
          "mode": "040700",
          "name": "/run/user/0",
          "obj_domain": "user_tmp_t",
          "obj_level": "s0",
          "obj_role": "object_r",
          "obj_user": "system_u",
          "objtype": "NORMAL",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        }
      ]
    }
  }
]
